lib/screens/search/states/search_state.dart

```diff
@@ -0,0 +1,12 @@
+import 'dart:core';
+
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'search_state.freezed.dart';
+
+@freezed
+class SearchState with _$SearchState {
+  const factory SearchState.loading() = _SearchStateLoading;
+  const factory SearchState.success() = _SearchStateSuccess;
```

Comment:
I imagine this part is just as a boilerplate so that in the future we can actually pass a real value when the state is Success for the case  of  the SearchScreen

Reply:
it's at the moment it is just a model